src/invidious/yt_backend/extractors.cr

```diff
@@ -435,20 +435,22 @@ private module Extractors
       raw_items = [] of JSON::Any
       content = extract_selected_tab(target["tabs"])["content"]
 
-      content["sectionListRenderer"]["contents"].as_a.each do |renderer_container|
-        renderer_container_contents = renderer_container["itemSectionRenderer"]["contents"][0]
-
-        # Category extraction
-        if items_container = renderer_container_contents["shelfRenderer"]?
-          raw_items << renderer_container_contents
-          next
-        elsif items_container = renderer_container_contents["gridRenderer"]?
-        else
-          items_container = renderer_container_contents
-        end
+      if content["sectionListRenderer"]
+        content["sectionListRenderer"]["contents"].as_a.each do |renderer_container|
```

Comment:
```suggestion
      if section_list_renderer = content["sectionListRenderer"]?
        section_list_renderer["contents"].as_a.each do |renderer_container|
```

Reply:
it's `[]?` to check if a hash key exists :)

Also, we've seen with Syeopite that the code runs faster when we create an intermediary variable inside the if condition!